### PR TITLE
Make runtime binaries not depend on usersim

### LINKED ISSUE #2756: ebpfapi.dll should not depend on usersim.dll
Non-test programs using eBPF for Windows should not pull in usersim.dll

### PR DESCRIPTION
**Pull request by dthaler**
## Description

* Make runtime binaries not depend on usersim
* Split libs\platform into libs\shared (used by both UM libraries and the eBPF runtime) and libs\runtime (used only by the eBPF runtime)
* Move some non-ebpf-specific functions like `ebpf_duplicate_string()`, `ebpf_utf8_string_free()`, etc. to the cxplat library
* Fix ebpf_result_t mapping to handle the 2 newest values

Fixes #2756

## Testing

Covered by existing tests

## Documentation

No impact

## Installation

No impact

### CODE REVIEW

**Review by Alan-Jowett — APPROVED** [review]

**Review by Alan-Jowett — APPROVED** [review]

File: libs/execution_context/ebpf_native.c [hunk 1]
```diff
@@ -1451,11 +1456,11 @@ _ebpf_native_clean_up_handle_cleanup_context(_Inout_ ebpf_native_handle_cleanup_
     }
 
     if (cleanup_context->handle_cleanup_work_item != NULL) {
-        // ebpf_free_preemptible_work_item() will free the handle_information.
-        ebpf_free_preemptible_work_item(cleanup_context->handle_cleanup_work_item);
-    } else {
-        ebpf_free(cleanup_context->handle_information);
+        cxplat_free_preemptible_work_item(cleanup_context->handle_cleanup_work_item);
+        cleanup_context->handle_cleanup_work_item = NULL;
```
Comment (gtrevi) [inline libs/execution_context/ebpf_native.c:1460]: Could we merge nulling the pointer into `cxplat_free_preemptible_work_item`?

File: libs/execution_context/ebpf_native.c [hunk 2]
```diff
@@ -1451,11 +1456,11 @@ _ebpf_native_clean_up_handle_cleanup_context(_Inout_ ebpf_native_handle_cleanup_
...
     }
+    ebpf_free(cleanup_context->handle_information);
+    cleanup_context->handle_information = NULL;
```
Comment (gtrevi) [inline libs/execution_context/ebpf_native.c:1463]: Same here, Could we merge nulling the pointer into `ebpf_free`?

File: libs/execution_context/ebpf_native.c (hunk 1 above)
Comment (dthaler) [reply libs/execution_context/ebpf_native.c:1460]: it'd be different from the normal POSIX convention.  @nibanks any opinion?

Comment (Alan-Jowett) [reply libs/execution_context/ebpf_native.c:1460]: Can we make this an issue and get this PR merged?

File: libs/execution_context/ebpf_native.c (hunk 2 above)
Comment (gtrevi) [reply libs/execution_context/ebpf_native.c:1463]: @Alan-Jowett @dthaler I'm fine to eventually to track this into another issue.

File: libs/execution_context/ebpf_native.c (hunk 1 above)
Comment (gtrevi) [reply libs/execution_context/ebpf_native.c:1460]: @Alan-Jowett @dthaler I'm fine to eventually to track this into another issue.

Comment (dthaler) [reply libs/execution_context/ebpf_native.c:1460]: It's not clear whether it's a good idea, but filed https://github.com/microsoft/usersim/issues/112 to track that discussion.

**Review by gtrevi — APPROVED** [review]

**Pull request merged**